src/pages/teachers/curriculum/[subjectPhaseSlug]/[tab].tsx

```diff
@@ -503,13 +503,65 @@ export function createDownloadsData(
   return downloadsData;
 }
 
```

Comment:
I think ideally this would be dealt with in the MV, but it also would have been too time consuming and complex to do that for this ticket.

This function ultimately filters the units to remove duplicates and retain only the most specific units. It identifies units with the same core slugs and checks if a more specific version exists by comparing optional fields (tier_slug, examboard_slug, pathway_slug). If a unit has null values for these optional fields while a more specific version has non-null values, the less specific unit is discarded. Additionally, it ensures only the first occurrence of each unique unit is kept in the final array.